CadRevealFbxProvider.Tests/Attributes/ScaffoldingCsvLineParserTests.cs

```diff
@@ -0,0 +1,184 @@
+        public int Index { get; set; } = 1;
+        public int ColumnCount => Headers.Length;
+        public string this[int index] => Values[index];
+        public string this[string header] => Values[Array.IndexOf(Headers, header)];
+    }
+
+    [Test]
+    public void ExtractColumnValueFromRow_WithValidHeader_ReturnsExpectedValue()
+    {
+        var row = new TestCsvLine { Headers = new[] { "A", "B", "C" }, Values = new[] { "1", "2", "3" } };
+        var result = ScaffoldingCsvLineParser.ExtractColumnValueFromRow("B", row);
+        Assert.That(result, Is.EqualTo("2"));
+    }
+
+    [Test]
+    public void ExtractSingleWeightFromCsvRow_WithValidWeightHeader_ReturnsWeight()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Weight kg", "Other" }, Values = new[] { "42", "foo" } };
+        var result = ScaffoldingCsvLineParser.ExtractSingleWeightFromCsvRow(row);
+        Assert.That(result, Is.EqualTo("42"));
+    }
+
+    [Test]
+    public void ExtractSingleWeightFromCsvRow_WithEmptyWeightHeader_ReturnsNull()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Weight kg", "Other" }, Values = new[] { "", "foo" } };
+        var result = ScaffoldingCsvLineParser.ExtractSingleWeightFromCsvRow(row);
+        Assert.That(result, Is.Null);
+    }
+
+    [Test]
+    public void ExtractSingleDescriptionFromCsvRow_WithManufacturerFalse_ReturnsDescriptionWithManufacturer()
+    {
+        var row = new TestCsvLine { Headers = new[] { "HAKI Description", "Other" }, Values = new[] { "desc", "foo" } };
+        var result = ScaffoldingCsvLineParser.ExtractSingleDescriptionFromCsvRow(row, false);
+        Assert.That(result, Does.Contain("desc"));
+    }
+
+    [Test]
+    public void ExtractKeyFromCsvRow_WithValidKeyHeader_ReturnsKey()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Item code", "Other" }, Values = new[] { "KEY123", "foo" } };
+        var result = ScaffoldingCsvLineParser.ExtractKeyFromCsvRow(row, 0, "Item code");
+        Assert.That(result, Is.EqualTo("KEY123"));
+    }
+
+    [Test]
+    public void ExtractKeyFromCsvRow_WithMissingKeyHeader_ThrowsException()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Item code", "Other" }, Values = new[] { "", "foo" } };
+        Assert.That(
+            () => ScaffoldingCsvLineParser.ExtractKeyFromCsvRow(row, 0, "Item code"),
+            Throws.TypeOf<ScaffoldingAttributeParsingException>()
+        );
+    }
+
+    [Test]
+    public void IsNumericSapColumn_WithNumericHeader_ReturnsTrue()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Work order", "Other" }, Values = new[] { "123", "foo" } };
+        var numericHeaders = new List<string> { "Work order" };
+        var result = ScaffoldingCsvLineParser.IsNumericSapColumn(row, 0, numericHeaders);
+        Assert.That(result, Is.True);
+    }
+
+    [Test]
+    public void IsNumericSapColumn_WithNonNumericHeader_ReturnsFalse()
```

Comment:
I am wondering about the goal of this test (trying to make sense of it given the test name only). Maybe we can take away the word "with", then it makes more sense to me :)

Reply:
Removed with prefix